### PR TITLE
Get actions from IElementActionProvider loaded via DependencyInjection

### PR DESCRIPTION
**Pull request by neexite**

### CODE REVIEW

File: Composite/C1Console/Elements/Foundation/ElementActionProviderFacade.cs [hunk 1]
```diff
@@ -326,6 +327,22 @@ public static void AddActions(IEnumerable<Element> elements, string providerName
                             Log.LogCritical("ElementActionProviderFacade", ex);
                         }
                     }
+
+                    foreach (var elementActionProviders in ServiceLocator.GetServices<IElementActionProvider>())
```
Comment (burningice2866) [inline Composite/C1Console/Elements/Foundation/ElementActionProviderFacade.cs:331]: Should be `elementActionProvider`  in singular

File: Composite/C1Console/Elements/Foundation/ElementActionProviderFacade.cs [hunk 2]
```diff
@@ -326,6 +327,22 @@ public static void AddActions(IEnumerable<Element> elements, string providerName
...
+                    {
+                        try
+                        {
+                            var actions = elementActionProviders.GetActions(element.ElementHandle.EntityToken);
+
+                            element.AddAction(actions);
+                        }
+                        catch (Exception ex)
+                        {
+                            Log.LogCritical("ElementActionProviderFacade",
```
Comment (burningice2866) [inline Composite/C1Console/Elements/Foundation/ElementActionProviderFacade.cs:341]: Can use nameof(ElementActionProviderFacade) instead of string

**Comment by burningice2866** [comment]
One challenge i found when injecting actions like this was controlling the order of actions; using composite.config its easy to control the order by adding providers before or after one another, but this way it becomes more random. Especially if various packages inject their services in their startup handler which isn't guaranteed to fire in any particular order if you have, say, 10 startup handlers.

Adding a way to control the order of providers here would be very helpful, in MEF you do it by annotating your component saying that it should be before or after another component by name

**Comment by neexite** [comment]
> One challenge i found when injecting actions like this was controlling the order of actions; using composite.config its easy to control the order by adding providers before or after one another, but this way it becomes more random. Especially if various packages inject their services in their startup handler which isn't guaranteed to fire in any particular order if you have, say, 10 startup handlers.
> 
> Adding a way to control the order of providers here would be very helpful, in MEF you do it by annotating your component saying that it should be before or after another component by name

Good point. Registering providers should be in ApplicationStartupHandler.ConfigureServices. I think we should add smth like weight for StartupHandler to sort the handlers. Because now it sorted by the result of Directory.GetFiles in Bin Directory.
@napernik What are you think?

**Pull request merged**